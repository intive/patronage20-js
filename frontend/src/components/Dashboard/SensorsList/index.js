import SensorsList from './SensorsList.jsx'

export default SensorsList
