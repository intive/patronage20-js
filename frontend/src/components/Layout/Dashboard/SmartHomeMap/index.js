import SmartHomeMap from './SmartHomeMap.jsx'

export default SmartHomeMap
