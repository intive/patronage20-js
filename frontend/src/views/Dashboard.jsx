import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { loadDashboard, changeSensorStatus } from '@data/actions/dashboard'
import Spinner from '../components/Spinner'

import Page404 from '../components/Page404'

function mapSensorsToList (sensors) { // only for debuging purposes, replace this once implementing real sensors list
  const dispatch = useDispatch()
  return sensors.map((sensor) => {
    return (
      <li
        key={sensor.id}
        onClick={() => {
          dispatch(changeSensorStatus(sensor.id))
        }}
      >
        <p>{sensor.type} </p>
      </li>
    )
  })
}

export default function Dashboard (props) {
  const dispatch = useDispatch()

  useEffect(() => {
    dispatch(loadDashboard())
  }, [])

  const {
    sensors,
    map,
    loadingSensors,
    loadingMap,
    error
  } = useSelector((state) => state.dashboard)

  let content = <Spinner />
  const loading = loadingSensors || loadingMap

  if (!loading) {
    if (!error) {
      const sensorsList = mapSensorsToList(sensors)
      content = (
        <div>
          <ul>
            {sensorsList}
          </ul>
          <h1>{map}</h1>
        </div>
      )
    } else {
      content = <Page404 />
    }
  }

  return (
    <>
      {content}
    </>
  )
}
