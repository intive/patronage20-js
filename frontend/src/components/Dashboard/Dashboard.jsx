import React, { useEffect } from 'react'
import { Grid } from '@material-ui/core'
import { useDispatch, useSelector } from 'react-redux'
import { loadSensors } from '@data/actions/sensor'

import SmartHomeMap from './SmartHomeMap'
import SensorsList from './SensorsList'
import Spinner from '../UI/Spinner'
import Page404 from '../UI/Page404'
import SensorsWarning from '../UI/Snackbars/SensorsWarning'

const Dashboard = () => {
  const dispatch = useDispatch()
  useEffect(() => {
    dispatch(loadSensors())
  }, [])

  const {
    loadingSensors,
    loadingError
  } = useSelector((state) => state.sensor)

  let content = <Spinner />

  if (!loadingSensors) {
    if (!loadingError) {
      content = (
        <Grid container maxwidth='xs' data-testid='dashboard-content-id'>
          <SensorsWarning />
          <SmartHomeMap />
          <SensorsList />
        </Grid>
      )
    } else {
      content = <Page404 />
    }
  }

  return (
    <>
      {content}
    </>
  )
}

export default Dashboard
