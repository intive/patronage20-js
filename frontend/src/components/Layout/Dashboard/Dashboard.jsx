import React, { useEffect } from 'react'
import { Grid } from '@material-ui/core'
import { useDispatch, useSelector } from 'react-redux'
import { loadSensors, changeSensorStatus } from '@data/actions/sensor'

import SmartHomeMap from './SmartHomeMap'
import SensorsList from './SensorsList'
import Spinner from '../../Spinner'
import Page404 from '../../Page404'

function mapSensorsToList (sensors) { // only for debuging purposes, replace this once implementing real sensors list
  const dispatch = useDispatch()
  return sensors.map((sensor) => {
    return (
      <li
        key={sensor.id}
        onClick={() => {
          dispatch(changeSensorStatus(sensor.id))
        }}
      >
        <p>{sensor.type} </p>
      </li>
    )
  })
}

const Dashboard = () => {
  const dispatch = useDispatch()
  useEffect(() => {
    dispatch(loadSensors())
  }, [])

  const {
    //sensors,
    loadingSensors,
    loadingError
  } = useSelector((state) => state.sensor)

  let content = <Spinner />

  if (!loadingSensors) {
    if (!loadingError) {
      // const sensorsList = mapSensorsToList(sensors)
      content = (
        <Grid container maxwidth='xs' data-testid='dashboard-content-id'>
          <SmartHomeMap />
          <SensorsList />
        </Grid>
      )
    } else {
      content = <Page404 />
    }
  }

  return (
    <>
      {content}
    </>
  )
}

export default Dashboard
