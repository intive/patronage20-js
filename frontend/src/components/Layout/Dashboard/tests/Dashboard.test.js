// /* globals , describe, it, expect */

import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import Dashboard from '../index'

import myStore from '../../../../store'

describe('<Dashboard />', () => {
  let component
  it('should render Dashboard with given store', () => {
    component = renderer.create(
      <Provider store={myStore}>
        <Dashboard />
      </Provider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})