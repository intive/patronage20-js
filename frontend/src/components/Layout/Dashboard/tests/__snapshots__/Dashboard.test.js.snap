// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should render Dashboard with given store 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
  data-testid="dashboard-content-id"
  maxwidth="xs"
>
  <div
    className="MuiGrid-root makeStyles-mapBackground-104 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-9"
  >
    <div
      className="makeStyles-mapContainer-105"
    >
       
      <img
        alt="Your Smart Home Map"
        src="/frontend/src/assets/changeMe.jpg"
      />
    </div>
  </div>
  <div
    className="MuiGrid-root makeStyles-listContainer-106 MuiGrid-item MuiGrid-grid-xs-3"
  />
</div>
`;
